validator/lib/utility.js

```diff
@@ -69,3 +72,36 @@ function sphereInsideSphere(sphereInner, sphereOuter) {
     var distance = Cartesian3.distance(centerInner, centerOuter);
     return distance <= (radiusOuter - radiusInner);
 }
+
+var scratchBoxCenter = new Cartesian3();
+var scratchSphereCenter = new Cartesian3();
+var scratchBoxHalfAxes = new Matrix3();
+
+function boxInsideSphere(box, sphere) {
+    var centerBox = Cartesian3.fromElements(box[0], box[1], box[2], scratchBoxCenter);
+    var halfAxesBox = Matrix3.fromArray(box, 3, scratchBoxHalfAxes);
+    var transformBox = Matrix4.fromRotationTranslation(halfAxesBox, centerBox);
+
+    var radiusSphere = sphere[3];
+    var centerSphere = Cartesian3.unpack(sphere, 0, scratchSphereCenter);
+
+    var cube = new Array(8);
+    cube[0] = new Cartesian3(-1, -1, -1);
+    cube[1] = new Cartesian3(-1, -1, 1);
+    cube[2] = new Cartesian3(1, -1, 1);
+    cube[3] = new Cartesian3(1, -1, -1);
+    cube[4] = new Cartesian3(-1, 1, -1);
+    cube[5] = new Cartesian3(-1, 1, 1);
+    cube[6] = new Cartesian3(1, 1, 1);
+    cube[7] = new Cartesian3(1, 1, -1);
+
+    var i = 0;
+    for (i = 0; i < 8; i++) {
```

Comment:
Same note from the other PR, this can now become `for (var i = 0`.

That's my only comment, all the other changes are solid.